packages/ui-testable/src/testable.ts

```diff
@@ -34,45 +35,57 @@ const testable =
   // ALWAYS_APPEND_UI_TESTABLE_LOCATORS=1
   // We do this because adding those `data-cid` locators slows things down.
   !process.env.ALWAYS_APPEND_UI_TESTABLE_LOCATORS
-    ? () => (Component) => Component
-    : decorator((ComposedComponent) => {
+    ? () => (Component: any) => Component
+    : decorator((ComposedComponent: any) => {
         const displayName =
           ComposedComponent.displayName || ComposedComponent.name
         const locator = {
           attribute: 'data-cid',
           value: displayName
         }
         const selector = `[${locator.attribute}~="${locator.value}"]`
-        class TestableComponent extends ComposedComponent {
+        class TestableComponent<
+          P = Record<string, unknown>,
+          S = ComponentState
+        > extends ComposedComponent {
           static selector = selector
-          componentDidMount(...args) {
+          private _testableUnmounted: boolean | undefined
+          _locatorTimeout: NodeJS.Timeout | undefined
+          private DOMNode: any
```

Comment:
This is also a tricky one. This variable is not defined here, its coming from the super class that `TestableComponent` extends. I'd either leave a TODO here to do this properly, or now make an interface that looks like this:
```
interface ClassWithDOMNode {
  DOMNode?: Element
}
```
and make the classes that have this prop (`ReactPortal` and `SubTreePortal`, maybe more implement it. 
In this class I'd make a cast, then there is no need to define a private property:
```node = (this as ClassWithDOMNode).DOMNode || findDOMNode(this)```

Reply:
I figured out the possible type of DOMNode, and typed it in the relevant components, but only added a TODO about the common interface. Would like to discuss how we want to do this in the future.